### ISSUE TITLE
Tasks with same name, in different components, get overwritten

### ISSUE THREAD

**Issue by tpetrone** [machty/ember-concurrency#349]
Steps to reproduce
- Create two components each `perform`_ing_ tasks with the same name when clicking a button.
- Render both components in the **same page**

What I am seeing:
- Clicking the button in the first component (appears sooner in the template) triggers the task of the second component
- Clicking the button in the second component also triggers the task declared in its own `js` file (the exact same task)

What I expected:
- Clicking the button in the first component (appears sooner in the template) triggers the task in its own `js` file
- Clicking the button in the second component triggers the task declared in its own `js` file (the exact same task)

In other words, the last component rendered is overwriting the task of the first one when both tasks have the exact same declaration. I.e.:
###### component 1
```js
   uploadFile: task(function * (file) {...}
```
###### component 2
```js
   uploadFile: task(function * (file) {...}
```
---
#### Is this an expected behavior?
---

**Comment by tpetrone** [comment]
I take it back. Further investigation elucidated that the problem was, in fact, related to how I was invoking `ember-file-upload` built-in component

Somehow related to:
https://github.com/adopted-ember-addons/ember-file-upload/issues/173
